frontend/src/concepts/modelRegistry/utils.ts

```diff
@@ -0,0 +1,36 @@
+export type ObjectStorageFields = {
+  endpoint: string;
+  bucket: string;
+  region?: string;
+  path: string;
+};
+
+export const objectStorageFieldsToUri = (fields: ObjectStorageFields): string => {
+  const { endpoint, bucket, region, path } = fields;
+  const searchParams = new URLSearchParams();
+  searchParams.set('endpoint', endpoint);
+  if (region) {
+    searchParams.set('defaultRegion', region);
+  }
+  return `s3://${bucket}/${path}?${searchParams.toString()}`;
+};
+
+export const uriToObjectStorageFields = (uri: string): ObjectStorageFields | null => {
+  try {
+    const urlObj = new URL(uri);
```

Comment:
I just told you that cause it strikes me that here we try catch to retrieve the fields but the method above we don't sanetize the inputs.

Reply:
Yeah the try-catch here was just specifically because the constructor for `new URL` can throw an error. It makes sense to check on both ends though.